tv/tv.cpp

```diff
@@ -326,8 +325,8 @@ struct TVLegacyPass final : public llvm::ModulePass {
         has_failure = true;
         *out << "\nPass: " << pass_name << '\n';
         emitCommandLine(out);
-        if (MClone)
-          writeBitcodeAtomically(report_filename);
+	if (SavedBitcode.size() > 0)
```

Comment:
This is writing the bitcode after the wrong transformation only. We need the bitcode before, which is pretty annoying. You would need to write it to disk all the times. And don't overwrite it later.

Reply:
please look closer, the call at line 329 is writing to disk some bitcode that we saved into a memory buffer (a std::string) earlier